cloudflare/data_source_zones.go

```diff
@@ -65,28 +65,23 @@
 		return err
 	}
 
-	zones, err := client.ListZones()
+	zoneFilter := cloudflare.WithZoneFilters(
+		fmt.Sprintf("contains:%s", filter.name),
```

Comment:
After sleeping on this, I might check if the domain is a full domain (`\w+.\w+` type of thing) and of it is, pass it without the `contains:` for an exact match.